common/src/main/java/org/conscrypt/ActiveSession.java

```diff
@@ -316,9 +321,13 @@ void onPeerCertificatesReceived(
      */
     void onHandshakeCompleted(String peerHost, int peerPort) {
         id = null;
-        this.peerHost = peerHost;
-        this.peerPort = peerPort;
         this.localCertificates = ssl.getLocalCertificates();
+        if (this.peerCertificates == null) {
+            // Under some as-yet-undetermined circumstances, the cert_verify_callback (which calls
```

Comment:
Unless there's some particular urgency, I think we'd feel better if we understood what these circumstances are before committing a workaround.